### PR TITLE
kernel: activate arch interface headers

### PR DESCRIPTION
**Pull request by tyrantlizard77**
These new headers are now pulled into the appropriate #include spaces.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
This documentation "harness" looks OK. I notice in the doc build artifacts (https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/19807/docs/guides/porting/arch.html) that under these section headings none of the API information is showing, but as you say, renaming functions with the excluded ``z_`` prefix is coming later.

**Review by carlescufi — APPROVED** [review]
LGTM, thanks

**Review by ioannisg — APPROVED** [review]
Need to see the posix failing errors, but otherwise, legitimate.

File: doc/zephyr.doxyfile.in
Comment (nashif) [inline doc/zephyr.doxyfile.in]: fix alignment here

Comment (andrewboie) [reply doc/zephyr.doxyfile.in]: I'll add this in another PR

**Comment by andrewboie** [comment]
@aescolar I had to touch some posix headers, please review.
I think there is more work to be done to fully disentangle things, but that can wait until we start work on #19666

**Review by nashif — APPROVED** [review]

**Comment by ioannisg** [comment]
Can we get this merged soon? And have sufficient time until the release to test possible issues with this refactoring.

**Comment by andrewboie** [comment]
@ioannisg waiting for @aescolar to review, I emailed him, once that's done I'll rebase for merge

File: arch/posix/include/posix_soc_if.h
Comment (aescolar) [inline arch/posix/include/posix_soc_if.h]: How come did you move these 2 back here? (it is not a deal breaker, but I would like to understand how come you decided to move those 2 into that file. And if need be place them in a more appropriate place)

**Review by aescolar — DISMISSED** [review]
Thank you very much @andrewboie for waiting until I was back, and giving me the opportunity to review it.
I'm ok with all posix arch related changed, except for that one comment below

File: arch/posix/include/posix_soc_if.h
Comment (andrewboie) [reply arch/posix/include/posix_soc_if.h]: @aescolar I moved these here because the definitions of these functions need to be where posix_irq_lock() and posix_irq_unlock() are defined.

Otherwise I have a lot of problems because of mutual dependencies. include/arch/posix/arch.h implicitly pulls in this header, but this header also implicitly pulls in include/arch/posix/arch.h due to including irq_offload.h, and since the arch interface header has prototypes to inline functions they must always be defined.

Comment (aescolar) [reply arch/posix/include/posix_soc_if.h]: Uhm.. if you want to get this in as is, then just dismiss my review. But I would after, (myself if nobody else), untangle this.

**Pull request merged**
